src/contexts/tokens.context.tsx

```diff
@@ -158,104 +154,107 @@ const TokensProvider: FC<PropsWithChildren> = (props) => {
+          const originalTokenChain = env.chains.find((chain) => chain.networkId === originNetwork);
+          if (!originalTokenChain) {
+            throw Error(`Could not find a chain that matched the originNetwork ${originNetwork}`);
+          }
+          return {
+            address: originTokenAddress,
+            chainId: originalTokenChain.chainId,
+            decimals,
+            logoURI,
+            name,
+            symbol,
+            wrappedToken: {
+              address,
+              chainId: chain.chainId,
+            },
+          };
+        })
+        .catch(() => {
+          // the provided address belongs to a native token
+          return addWrappedToken({
+            token: {
+              address,
+              chainId: chain.chainId,
+              decimals,
+              logoURI,
+              name,
+              symbol,
+            },
+          });
+        });
     },
-    [addWrappedToken]
+    [addWrappedToken, env, getNativeTokenInfo]
   );
 
   const getToken = useCallback(
-    async ({ chain, env, originNetwork, tokenAddress }: GetTokenParams): Promise<Token> => {
+    async ({ env, originNetwork, tokenOriginAddress }: GetTokenParams): Promise<Token> => {
+      const chain = env.chains.find((chain) => chain.networkId === originNetwork);
+      if (!chain) {
+        throw new Error(
+          `The chain with the originNetwork "${originNetwork}" could not be found in the list of supported Chains`
+        );
+      }
       const token = [...getCustomTokens(), ...(tokens || [getEtherToken(chain)])].find(
         (token) =>
-          token.address === tokenAddress ||
-          (token.wrappedToken && token.wrappedToken.address === tokenAddress)
+          (token.address === tokenOriginAddress && token.chainId === chain.chainId) ||
+          (token.wrappedToken &&
+            token.wrappedToken.address === tokenOriginAddress &&
+            token.wrappedToken.chainId === chain.chainId)
       );
 
       if (token) {
         return token;
       } else {
-        const chain = env.chains.find((chain) => chain.networkId === originNetwork);
-
-        if (chain) {
-          return await getTokenFromAddress({ address: tokenAddress, chain }).catch(() => {
-            throw new Error(
-              `The token with the address "${tokenAddress}" could not be found either in the list of supported Tokens or in the blockchain with network id "${originNetwork}"`
-            );
-          });
-        } else {
+        return getTokenFromAddress({ address: tokenOriginAddress, chain }).catch(() => {
           throw new Error(
-            `The token with the address "${tokenAddress}" could not be found in the list of supported Tokens and the provided network id "${originNetwork}" is not supported`
+            `The token with the address "${tokenOriginAddress}" could not be found either in the list of supported Tokens or in the blockchain with network id "${originNetwork}"`
```

Comment:
The network id is something special from the zkEVM, I would use the Chain Id here

Reply:
Done